### PR TITLE
limit rss to 10 & use standard hugo limit var

### PR DESCRIPTION
**Pull request by dradetsky**
First, the k8s rss feed is too big. It's ~800kb every time I run a feed check, which can be problematic in certain circumstances, like if I haven't (or can't) configure my reader to check feeds in parallel, or I just have slow internet. And it extends back to midway through last year. A user almost always wants to catch up the last few articles she might have missed, not populate a local copy of the last 1-2 years worth of k8s blog posts. So we reduce the number of items emitted from 50 to 10.

Second, let's say in the future somebody said "We should actually bump it back up to 20 items" (not an unreasonable compromise). He would probably go looking in the Hugo docs and see the `rssLimit` variable documented and attempt to set it, and be confused as to why it wasn't changing (or wasn't already set). So we change the template to use this variable in order to be Hugo-idiomatic.

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
I'm wary of dropping the article limit much below 50.

Maybe we could make the images smaller instead?

File: config.toml [hunk 1]
```diff
@@ -33,6 +33,8 @@ enableGitInfo = true
 # Hindi is disabled because it's currently in development.
 disableLanguages = ["hi", "no"]
 
+rssLimit = 10
```
[the remark this answers is not in the document]
Comment (dradetsky) [reply config.toml:36]: I don't think it would be that puzzling, since almost all feeds are limited to some extent. But if you mean I add the feed 30 days after the release, and then I only get 10 of the 35 articles about the new release, I definitely don't think that'd be puzzling, since I probably just saw the list off all 30 articles in the blog while I was clicking the RSS link, so I clearly see that it's the most recent 10. And since I found the RSS link by looking at the blog on the web, I necessarily know how to see earlier articles if I want them.

This may be a matter of perspective, but I think that the main use case of RSS is almost always to subscribe. That is, I'm using it to pick up the last article or two every day, or every hour, or every 10 min. I'm not using it to populate a local copy of the last year's worth of posts. I don't really need that. Or at least, I don't need it nearly as much as I need to subscribe.

Making the images smaller might be a good idea for other reasons, but in this case I'm only thinking about the size of the feed as plain text. 800kb isn't very big for one feed, but I might have 200 feeds. I don't think it's a good idea to worsen the experience for subscribers and heavy users for the sake of people who want to get a local copy of the blog in their reader. However, if you think the latter group are important to cater to, I would think having some extra feed urls would be the way to do it. You have the main feed for the primary use case (subscribers keeping up to date on the latest 1-2 articles) and then a link to a page of alternate feeds, such as a feed of the entire blog since the beginning of time, or other formats (there's this jsonfeed thing too now apparently), or other things I can't think of.

But I would just do the 1 feed for the 1 case that IMO actually matters if it was me. Or at least, that's what I'd do _first_.

[the remark this answers is not in the document]
Comment (sftim) [reply config.toml:36]: For what it's worth, the feed is 200KiB gzipped. https://k8s.io/blog/ is 800KiB, too.

OK, I'm convinced.

```suggestion
# Limit number of blog articles included in the syndication feed
[services.rss]
limit = 10
```

If folks complain about the limited number of articles in the feed, we can provide a separate “last 2 years” feed or similar.

[the remark this answers is not in the document]
Comment (sftim) [reply config.toml:36]: I still think that smaller images would shrink the feed more, mind. The “plain text” of the feed is actually XML with embedded images.

[the remark this answers is not in the document]
Comment (dradetsky) [reply config.toml]: By "embedded images" do you mean html links to images, or something which directly encodes images in text, like svg or whatever? If the former, I assume we just have to live with that. If the latter, that could be a concern, but while I can't say I looked through the feed text super-carefully, I don't think I noticed any substantial amount of that. In any case, even if it was there, it's not obvious what we could do about it, or what the consequences of these options might be.

In any case, as I say, I didn't notice a whole lot of either kind of image. There were some big blocks of what I think where html code blocks which had been escaped, which do contribute nontrivial bloat. But once again I'm not totally sure we have easy options for getting rid of that bloat with a 100% guarantee of no unintended consequences. And there are plenty of articles which are bigger than the articles "bloated" by code blocks which are just big with content, like the Box/Laverack interview. So if you had a bunch of articles like that you'd be paying for the complexity of whatever you did to reduce the bloat from code blocks, but having just as bloated a feed.

This is why I wanted to solve the issue by making the feed smaller.

**Comment by sftim** [comment]
BTW, the image embedding is this code (a few lines below, in the same file):
```hugo
        {{ $img := (.Resources.ByType "image").GetMatch "*featured*" }}
        {{ with $img }}
        {{ $img := .Resize "640x" }}
        {{ printf "<![CDATA[<img src=\"%s\" width=\"%d\" height=\"%d\"/>]]>" $img.Permalink $img.Width $img.Height | safeHTML }}
        {{ end }}
```

**Comment by sftim** [comment]
If articles had _summaries_ we could put those in the feed, rather than the text of the entire article.

**Comment by dradetsky** [comment]
@sftim as contradictory as it may seem given my concern with feed size, I personally prefer whole-article feeds for various reasons.

In any case, the trouble with summaries is that either they're handwritten (meaning more work for authors, which I assume we want to avoid), or they're autogenerated in which you get all kinds of fun & entertaining edge cases, like unclosed html tags, or the weird results of stripping html tags from text. But even if we had a solution for summaries which we had very high confidence had none of these issues (unlikely IMO), there would still be trade-offs.

For example, some people might want to read at least some articles straight from their feed reader. Personally, I have feeds of service statuses, so that when somebody reports an issue with some internal service I can quickly rule out that it's not an issue with Github Webhooks again, and I just read these from the reader. If you're sending the whole article, you don't need to think about how your users want to use the feed content and their reader; they'll follow the feed item to the original article if they want.

Basically, if you can make the feed workable with whole-articles, I think that's best. And IMO we can.

**Comment by dradetsky** [comment]
@sftim it sounds like you fundamentally approved this. Are any other steps necessary to actually merge it?

**Comment by dradetsky** [comment]
Can a reviewer be assigned to this please?

**Comment by sftim** [comment]
I have already had a look at this PR. There is at least one [higher-priority PR](https://github.com/kubernetes/website/pulls?q=is%3Apr+is%3Aopen+label%3Apriority%2Fcritical-urgent%2Cpriority%2Fimportant-soon%2Cpriority%2Fimportant-longterm) awaiting review; if I were going to make a special callout within SIG Docs, I'd prefer to focus on those other PRs. Once there are no high priority PRs that are awaiting review, we could revisit.

**Comment by dradetsky** [comment]
@sftim according to that link, there aren't any anymore.

**Comment by divya-mohan0209** [comment]
@dradetsky : Thank you for this PR and for your patience. I think it'd be worthwhile adding to this agenda of the next SIG Docs meetings (EU/US & APAC) for wider awareness & discussions.

**Comment by kbhawkey** [comment]
👋 @dradetsky .
Before moving forward, I suggest to create an issue that describes the proposed changes.

**Comment by divya-mohan0209** [comment]
Hello @dradetsky , I'd be happy to make a callout within SIG Docs by adding this to the agenda item for the upcoming week's meeting. Please could you rebase the PR before Tuesday?

**Comment by divya-mohan0209** [comment]
Hello @dradetsky , since there has been no response from your side for over 2 months we'll be closing off this PR currently. Should you have any free cycles in the future, please do feel free to reopen it by issuing the /reopen command.

We'd, however, request to consider your bandwidth for the following before reopening the PR:

- Create an associated issue
- Bring the issue & the PR up during one of the meetings for further discussion.

Thanks!

/close
